server/image.go

```diff
@@ -189,9 +198,55 @@ func (im *Image) Release() {
 	}
 }
 
-func (im *Image) sync() {
+func (im *Image) sync() error {
 	im.Width = im.img.Width()
 	im.Height = im.img.Height()
+
+	if im.Format == "mp4" && len(im.Data) != 0 {
+		// No need to convert again.
+		return nil
+	}
+
 	im.Format = im.img.Format()
 	im.Data = im.img.Data()
+
+	if im.conversionFormat == "mp4" && im.Format == "gif" {
```

Comment:
Handling the special case from `gif` to `mp4`. We currently don't have a way to convert our GIF data stream into a MP4 data stream, so we need to output to disk both files. We may be able to pipe the GIF bytes directly into ffmpeg, the docs say it would be something like: `ffmpeg -f image2pipe - -y output.mp4`, but that command does not work right away, some parameter is missing.